### PR TITLE
Add max button to transfer

### PR DESCRIPTION
**Pull request by cehali**
# Summary
Add max button to transfer

### CODE REVIEW

**Comment by cehali** [comment]
image
image

File: universal-login-wallet/test/integration/ui/Transfer.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ describe('UI: Transfer', () => {
       : '0x0';
   });
 
-  it('send ETH => invalid ensName', async () => {
+  xit('send ETH => invalid ensName', async () => {
```
Comment (marekkirejczyk) [inline universal-login-wallet/test/integration/ui/Transfer.tsx]: `xit` -> `it`

File: universal-login-sdk/lib/core/utils/isBiggerThan.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import {utils} from 'ethers';
+
+export const isBiggerThan = (value: utils.BigNumber, validationValue: utils.BigNumber) => value.gt(validationValue) ? value : validationValue;
```
Comment (xitronix) [inline universal-login-sdk/lib/core/utils/isBiggerThan.ts]: 'Is' suggest that return type will be boolean. Maybe it should be getBigger?

File: universal-login-sdk/lib/core/services/TransferService.ts [hunk 3]
```diff
@@ -16,13 +21,34 @@ export class TransferService {
...
   validateInputs(transferDetails: TransferDetails, balance: Nullable<string>) {
+    this.errors = {amount: [], to: []};
     const {gasPrice, gasToken} = transferDetails.gasParameters;
     const gasCostInWei = utils.bigNumberify(DEFAULT_GAS_LIMIT.toString()).mul(gasPrice);
     ensureNotNull(balance, Error, 'Balance is null');
     const isAmountValid = isValidAmount(transferDetails.transferToken, balance, gasToken, gasCostInWei, transferDetails.amount);
+    if (!isAmountValid) {
+      this.errors.amount.push(`Amount ${transferDetails.amount} is not valid`);
+    }
     const isRecipientValid = isValidRecipient(transferDetails.to);
```
Comment (xitronix) [inline universal-login-sdk/lib/core/services/TransferService.ts]: inline isAmountValid and  isRecipientValid. I think we don't loose code readability:
```ts
if (!isValidAmount(.....)) {

}
```

File: universal-login-sdk/lib/core/utils/isBiggerThan.ts (hunk 2 above)
Comment (JustynaBroniszewska) [reply universal-login-sdk/lib/core/utils/isBiggerThan.ts]: 👍

File: universal-login-sdk/lib/core/services/TransferService.ts [hunk 4]
```diff
@@ -16,13 +21,34 @@ export class TransferService {
...
+
+  getMaxAmount(gasParameters: GasParameters, balance: Nullable<string>) {
+    ensureNotNull(balance, Error, 'Balance is null');
+    const {gasPrice, gasToken} = gasParameters;
+    if (gasToken !== ETHER_NATIVE_TOKEN.address) {
+      return balance;
+    }
+    const gasCostInWei = utils.bigNumberify(DEFAULT_GAS_LIMIT.toString()).mul(gasPrice);
+    const maxAmountAsBigNumber = utils.parseEther(balance).sub(gasCostInWei);
+    const maxAmountValidated = isBiggerThan(maxAmountAsBigNumber, utils.parseEther('0'));
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/lib/core/services/TransferService.ts]: `utils.parseEther('0')` is still 0.

File: universal-login-sdk/lib/core/utils/isBiggerThan.ts (hunk 2 above)
Comment (cehali) [reply universal-login-sdk/lib/core/utils/isBiggerThan.ts]: 👍

File: universal-login-sdk/lib/core/services/TransferService.ts (hunk 4 above)
Comment (cehali) [reply universal-login-sdk/lib/core/services/TransferService.ts]: But 0 is not BigNumber which is required by isBiggerThan (getBigger).

File: universal-login-sdk/lib/core/utils/isBiggerThan.ts (hunk 2 above)
Comment (marekkirejczyk) [inline universal-login-sdk/lib/core/utils/isBiggerThan.ts]: bigNumberMax()

File: universal-login-sdk/test/unit/integration/TransferService.ts [hunk 5]
```diff
@@ -107,43 +107,48 @@ describe('UNIT: TransferService', () => {
...
-    expect(() => transferService.validateInputs({
-      to: '0x',
-      amount: '123',
+    expect(transferService.validateInputs({
+      to: TEST_ACCOUNT_ADDRESS,
+      amount: '350',
       transferToken: ETHER_NATIVE_TOKEN.address,
       gasParameters,
-    }, balance)).to.throw('0x is not valid');
+    }, balance).amount[0]).to.eq('Amount 350 is not valid');
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/unit/integration/TransferService.ts]:
```ts
    }, balance))
.to.deep.eq({amount: ['Amount 350 is not valid']});
```

**Pull request merged**